format_context.go

```diff
@@ -106,6 +106,13 @@ func (fc *FormatContext) Metadata() *Dictionary {
 	return newDictionaryFromC(fc.c.metadata)
 }
 
+func (fc *FormatContext) Metadata(d *Dictionary) {
+	if d == nil {
```

Comment:
I'd rather not ignore if `d == nil` in case someone would want to reset the metadata to `nil`